tidy3d/components/grid/grid_spec.py

```diff
@@ -665,14 +787,580 @@ def _preprocessed_structures(self, structures: List[StructureType]) -> List[Stru
+
+    Note
+    ----
+
+    This class only takes effect when :class:`AutoGrid` is applied.
+
+    Example
+    -------
+    >>> layer_spec = LayerRefinementSpec(axis=2, center=(0,0,0), size=(td.inf, td.inf, 1))
+
+    """
+
+    axis: Axis = pd.Field(
+        ...,
+        title="Axis",
+        description="Specifies dimension of the layer normal axis (0,1,2) -> (x,y,z).",
+    )
+
+    min_steps_along_axis: Optional[pd.PositiveFloat] = pd.Field(
+        None,
+        title="Minimal Number Of Steps Along Axis",
+        description="If not ``None`` and the thickness of the layer is nonzero, set minimal "
+        "number of steps discretizing the layer thickness.",
+    )
+
+    bounds_refinement: Optional[GridRefinement] = pd.Field(
+        None,
+        title="Mesh Refinement Factor Around Layer Bounds",
+        description="If not ``None``, refine mesh around minimum and maximum positions "
+        "of the layer along axis dimension. If `min_steps_along_axis` is also specified, "
+        "refinement here is only applied if it sets a smaller grid size.",
+    )
+
+    bounds_snapping: Optional[Literal["bounds", "lower", "upper", "center"]] = pd.Field(
+        "lower",
+        title="Placing Grid Snapping Point Along Axis",
+        description="If not ``None``, enforcing grid boundaries to pass through ``lower``, "
+        "``center``, or ``upper`` position of the layer; or both ``lower`` and ``upper`` with ``bounds``.",
+    )
+
+    corner_finder: Optional[CornerFinderSpec] = pd.Field(
+        CornerFinderSpec(),
+        title="Inplane Corner Detection Specification",
+        description="Specification for inplane corner detection. Inplane mesh refinement "
+        "is based on the coordinates of those corners.",
+    )
+
+    corner_snapping: bool = pd.Field(
+        True,
+        title="Placing Grid Snapping Point At Corners",
+        description="If ``True`` and ``corner_finder`` is not ``None``, enforcing inplane "
+        "grid boundaries to pass through corners of geometries specified by ``corner_finder``.",
+    )
+
+    corner_refinement: Optional[GridRefinement] = pd.Field(
+        GridRefinement(),
+        title="Inplane Mesh Refinement Factor Around Corners",
+        description="If not ``None`` and ``corner_finder`` is not ``None``, refine mesh around "
+        "corners of geometries  specified by ``corner_finder``. "
+        "This option is similar to edge refinement if the edge is axis-aligned.",
+    )
+
+    @pd.validator("axis", always=True)
+    @skip_if_fields_missing(["size"])
+    def _finite_size_along_axis(cls, val, values):
+        """size must be finite along axis."""
+        if np.isinf(values["size"][val]):
+            raise SetupError("'size' must take finite values along 'axis' dimension.")
+        return val
+
+    @classmethod
+    def from_unbounded(
+        cls,
+        axis: Axis,
+        bounds: Tuple[float, float],
```

Comment:
This is a bit nitpicky perhaps but the usage `from_unbounded(bounds=...)` struck me as a bit strange. It is not immediately clear what's unbounded and what is bounded. Do you think it would make sense to call it e.g. `from_inplane_unbounded` and even maybe `layer_bounds` instead of `bounds`?

Reply:
Agree, I also feel it reads strange. I like `layer_bounds`!